src/core/execute.c

```diff
@@ -1851,6 +1851,40 @@ char** exec_context_get_restrict_filesystems(const ExecContext *c) {
         return l ? TAKE_PTR(l) : strv_new(NULL);
 }
 
+static int extension_name_is_vpicked(const char *path) {
+        _cleanup_free_ char *dir = NULL;
+        int r;
+
+        assert(path);
+
+        r = path_extract_filename(path, &dir);
+        if (r == -EADDRNOTAVAIL)
+                return 0;
+        if (r < 0)
+                return r;
+
+        return !!endswith(dir, ".v");
```

Comment:
hmm, so strictly speaking we support two syntaxes for .v/ dirs. one is:

`...PATH/NAME.SUFFIX.v/`

the other is:

`......PATH.v/NAME___.SUFFIX`

you only detect the former here, the latter should be detected too.

see systemd.v(7) man page, at the top.

the est hence should either check if either the final component or the 2nd to last compnet of the path has the .v/ suffix.

i think this helper should be moved to src/shared/vpick.[ch] btw, and get a test.

Reply:
This would be good to have, thanks. `path_uses_vpick` added in separate commit, with tests.